File: backend/auth-service/src/main/java/com/tuanhust/authservice/service/AuthService.java
package com.tuanhust.authservice.service;

import com.tuanhust.authservice.config.UserPrincipal;
import com.tuanhust.authservice.repsonse.AuthResponse;
import com.tuanhust.authservice.request.CreatePassword;
import com.tuanhust.authservice.request.LoginRequest;
import com.tuanhust.authservice.request.UpdatePassword;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.Valid;


public interface AuthService {
    AuthResponse login(LoginRequest loginRequest, HttpServletRequest request);
    AuthResponse refreshToken(String refreshToken,String accessToken);
    void logout(UserPrincipal userPrincipal);
    UserPrincipal verifyToken(String accessToken);
    AuthResponse loginWithGoogle(String token, HttpServletRequest request);
    void createPassword(String userId, CreatePassword createPassword);

    void updatePassword(String userId, @Valid UpdatePassword updatePassword);
}

File: fontend/task-management-fe/src/features/auth/services/authService.ts
import type {AuthResponse, LoginRequest} from "../types/auth.types"
import {privateApi, publicApi} from "@config/api.config"


export const AuthService = {
    login: async (data: LoginRequest): Promise<AuthResponse> => {
        const response = await publicApi.post<AuthResponse>('/auth/login', data);
        return response.data;
    },
    loginWithGoogle: async (token: string): Promise<AuthResponse> => {
        const response = await publicApi.post<AuthResponse>('/auth/google', {token: token});
        return response.data;
    },
    logout: async (): Promise<void> => {
        return await privateApi.post<void>('/auth/logout');
    },
    createPassword: async (data: {password:string,confirmPassword:string}): Promise<void> => {
        return await privateApi.post<AuthResponse>('/auth/password', data);
    },
    updatePassword: async (data: {password:string,newPassword:string,confirmPassword:string}): Promise<void> => {
        return await privateApi.put<AuthResponse>('/auth/password', data);
    },
}